pyxem/signals/correlation1d.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2022 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+
+
+from hyperspy.signals import BaseSignal, Signal1D
+import numpy as np
+from fractions import Fraction as frac
+from pyxem.utils.correlation_utils import _get_interpolation_matrix, _symmetry_stem
+
+
+class Correlation1D(Signal1D):
+    """Signal class for pearson correlation and symmetry coefficient."""
+    _signal_type = "correlation"
+
+    def __init__(self, *args, **kwargs):
+        """Create a Correlation 1D signal from a numpy.ndarray.
+
+        Parameters
+        ----------
+        *args :
+            Passed to the __init__ of Signal1D. The first arg should be
+            a numpy.ndarray
+        **kwargs :
+            Passed to the __init__ of Signal1D
+        """
+        super().__init__(*args, **kwargs)
+
+        self.decomposition.__func__.__doc__ = BaseSignal.decomposition.__doc__
+
+    def get_symmetry_coefficient(self,
+                                 angular_range=0.1,
+                                 symmetries=[2, 3, 4, 5, 6, 7, 8, 9, 10],
+                                 method="average",
+                                 include_duplicates=False,
+                                 normalize=True,
+                                 ):
+        """Return symmetry coefficient from pearson correlation function at all real
+        space positions (n from 2 to 10).
+
+        Parameters
+        ----------
+        angular_range: float
+            The angular range (in rad) to integrate over when calculating the symmetry coefficient.
+        symmetries: list
+            The list of symmetries to calculate.
+        method: "average", "first", "max"
+            The method for calculating the Symmetry STEM
+        include_duplicates: bool
+            If angles which are duplicated should be included.
+        normalize: bool
+            This normalized by dividing by the number of angles for each symmetry.
+
+        Returns
+        -------
+        sn: Signal1D
+            Symmetry coefficient
+        """
+        angles = [set(frac(j, i) for j in range(0, i)) for i in symmetries]
```

Comment:
Sampling of symmetry coefficients start from the first non-zero angle, since Pearson correlation at zero shift is always unity. So I think the angles of each symmetry should be `[set(frac(j, i) for j in range(1, i)) for i in symmetries]`

Reply:
That is a good point.  Do I understand correctly that the zero shift is always 1?  Is there any case where it might be useful to return the zero shift for a basis of comparison or not?   I'm not sure what the best way to deal with the case where you want to include/ not include duplicate angles.

Reply:
Yes, zero shift is always 1 based on the way we normalize it (which is intended), so I can't think of a case where it'd be useful to include it. And I think you handle duplicate angles really well. I don't think duplicate angles option would interfere with any other parameters once you exclude the zero shift.

Reply:
Sounds good! I've removed the 0 shift term from the symmetry stem